internal/pkg/v1/action/operator_update.go

```diff
@@ -0,0 +1,106 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/blang/semver/v4"
+	olmv1 "github.com/operator-framework/operator-controller/api/v1"
+
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+type OperatorUpdate struct {
+	config *action.Configuration
+
+	Package string
+
+	Version                 string
+	Channels                []string
+	UpgradeConstraintPolicy string
+	OverrideUnset           bool
+
+	CleanupTimeout time.Duration
+
+	Logf func(string, ...interface{})
+}
+
+func NewOperatorUpdate(cfg *action.Configuration) *OperatorUpdate {
+	return &OperatorUpdate{
+		config: cfg,
+		Logf:   func(string, ...interface{}) {},
+	}
+}
+
+func (ou *OperatorUpdate) Run(ctx context.Context) (*olmv1.ClusterExtension, error) {
+	var ext olmv1.ClusterExtension
+
+	opKey := types.NamespacedName{Name: ou.Package}
+	if err := ou.config.Client.Get(ctx, opKey, &ext); err != nil {
+		return nil, err
+	}
+
+	if ext.Spec.Source.SourceType != olmv1.SourceTypeCatalog {
+		return nil, fmt.Errorf("unrecognized source type: %q", ext.Spec.Source.SourceType)
+	}
+
+	ou.setDefaults(ext.Spec.Source.Catalog)
+	constraintPolicy := olmv1.UpgradeConstraintPolicy(ou.UpgradeConstraintPolicy)
+	if !ou.needsUpdate(ext.Spec.Source.Catalog, constraintPolicy) {
+		return nil, errNoChange
+	}
+
+	if ou.Version != "" {
+		if _, err := semver.ParseRange(ou.Version); err != nil {
+			return nil, fmt.Errorf("failed parsing version: %w", err)
+		}
+	}
+
+	ext.Spec.Source.Catalog.Version = ou.Version
+	ext.Spec.Source.Catalog.Channels = ou.Channels
+	ext.Spec.Source.Catalog.UpgradeConstraintPolicy = constraintPolicy
+	if err := ou.config.Client.Update(ctx, &ext); err != nil {
+		return nil, err
+	}
+
+	if err := waitUntilOperatorStatusCondition(ctx, ou.config.Client, &ext, olmv1.TypeInstalled, metav1.ConditionTrue); err != nil {
+		return nil, fmt.Errorf("timed out waiting for operator: %w", err)
+	}
```

Comment:
no cleanup is what olmv0 does as well. I think we agreed to ignore approach to waiting/cleanup here: https://github.com/operator-framework/kubectl-operator/pull/220#discussion_r1973356442, but still wanted to call it out here explicitly

Reply:
Is this comment sorted out?
Can we close it to make it easier for others to follow up?
What should I look at?

Reply:
I thought it would be useful to reviewers

Reply:
I'm just trying to understand what you mean without getting pulled into other threads and multiple links since you have the full context.

So, wdyt adding a brief comment in the code with a simple overview/explanation (KISS) to make it clear for everyone if that is important?

Reply:
I followed the threads, so what about add on top of the method:

```
// The CLI should ideally provide status updates to indicate progress or recoverable issues,  
// but for now, maintaining the current wait behavior makes sense. A broader rethink of wait logic  
// across the CLI may be needed, potentially with a flag to control waiting behavior.  
// This relates to handling the `Progressing` condition and retry limits in the API.  
// More info: https://github.com/operator-framework/operator-controller/issues/1440
```

Reply:
At this point, I wouldn't say we know what the `ideal` scenario here is (https://github.com/operator-framework/kubectl-operator/pull/225#issuecomment-2710087085), but I don't think it needs to be a code comment. I'll still add it on the epic issue to keep track when we get to those discussions.
Hope that's ok.